src/dmd/traits.d

```diff
@@ -412,6 +413,31 @@ extern (C++) Expression pointerBitmap(TraitsExp e)
     return ale;
 }
 
+/**
+* __traits(getCallerSource, symbol) => get symbol source code (string[T.length] or string)
+*/
+extern (C++) Expression traitsGetSource(TraitsExp e)
+{
+    if (!e.args || e.args.dim < 1|| e.args.dim > 2)
+    {
+        error(e.loc, "TODO");
+        return new ErrorExp();
+    }
+    auto ret = new ArgStringInitExp(e.loc);
+    auto ident_expr=(*e.args)[0];
```

Comment:
calling it `arg0` would make more sense, especially since the next line verifies that it is a type, not an identifier.